openstack/identity/v3/extensions/trust/request.go

```diff
@@ -7,34 +7,35 @@ import (
 )
 
 type AuthOptionsExt struct {
-  token3.AuthOptions
-  TrustID string
+	token3.AuthOptions
+	TrustID string
 }
 
 func (ao AuthOptionsExt) ToAuthOptionsV3Map(c *gophercloud.ServiceClient, scope *token3.Scope) (map[string]interface{}, error) {
 	//Passing scope value to nil to add scope later in this function.
 	authMap, err := ao.AuthOptions.ToAuthOptionsV3Map(c, nil)
-        if err != nil {
-                return nil, err
-        }
+	if err != nil {
+		return nil, err
+	}
+	authMap = authMap["auth"].(map[string]interface{})
 
 	// Add a "scope" element if a Scope has been provided.
 	if ao.TrustID != "" {
-			// TrustID provided.
-			authMap["scope"] = map[string]interface{}{
-				"OS-TRUST:trust": map[string]interface{}{
-					"id" : ao.TrustID,
-					},
-				}
-		} else {
-			return nil, token3.ErrScopeEmpty
+		// TrustID provided.
+		authMap["scope"] = map[string]interface{}{
+			"OS-TRUST:trust": map[string]interface{}{
+				"id": ao.TrustID,
+			},
+		}
+	} else {
+		return nil, token3.ErrScopeEmpty
 	}
-  return authMap, nil
+	return map[string]interface{}{"auth": authMap}, nil
```

Comment:
hum however I think we can safely remove the explicit cast on this line indeed. I'll try and push if it compiles.